go/exorelay/main.go

```diff
@@ -1,10 +1,49 @@
 package exorelay
 
-type ExoRelay struct{}
+import (
+	"encoding/json"
 
+	"github.com/Originate/exocom/go/structs"
+
+	"golang.org/x/net/websocket"
+)
+
+// ExoRelay is the Go API to talk to Exocom
+type ExoRelay struct {
+	url    string
+	socket *websocket.Conn
+}
+
+// New creates a new ExoRelay instance
 func New(url string, config map[string]interface{}) *ExoRelay {
-	return new(ExoRelay)
+	return &ExoRelay{
+		url: url,
+	}
 }
 
-func (*ExoRelay) Connect() {
+// Connect brings an ExoRelay instance online
+func (exoRelay *ExoRelay) Connect() error {
+	socket, err := websocket.Dial(exoRelay.url, "", "origin:")
+	if err != nil {
+		return err
+	}
+	exoRelay.socket = socket
+	err = exoRelay.Send("exocom.register-service", nil)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
I'm not sure which is more idiomatic, but we should stick to one style. Later on we return the error directly so this should either be
```go
return exoRelay.Send("exocom.register-service", nil)
```

or we should add the error handling shenanigans below